### PR TITLE
Fixes for QnA / Recognizer Adaptive telemetry

### LINKED ISSUE #3472: QnAMakerDialog should create QnAMaker with more parameters
### Use this [query](https://github.com/Microsoft/botbuilder-dotnet/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen++label%3Aenhancement) to search for the most popular feature requests.

**Is your feature request related to a problem? Please describe.**

QnAMakerDialog creates QnAMaker without parameters like IBotTelemetryClient telemetryClient, bool logPersonalInformation. So we could not control these.

**Describe the solution you'd like**

Could configure all parameters for QnAMaker.

**Describe alternatives you've considered**

**Additional context**

https://github.com/microsoft/botbuilder-dotnet/blob/5c71a77a732d33d15e84929e429c5104f63b477f/libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog.cs#L143

[enhancement]

### PR DESCRIPTION
**Pull request by garypretty**
- Adds UseTelemetry extension method for DiaogManager which adds IBotTelemetryClient to turn state for use by adaptive recognizers.
 - Adaptive recognizers will now use the IBotTelemetryClient on the turn state unless the TelemetryClient property on the recognizer has been set.
 - Adds LogPersonalInformation property to QnAMaker dialogs and LUIS / QnA adaptive recognizers. Adaptive LUIS / QnA will look in settings.telemetry.logPersonalInformation by default.
 - Ensures QnAMaker client is always passed a TelemetryClient if set, along with the LogPersonalInformation flag (Fixes #3472)
 - LUIS Adaptive recognizer: Ensure telemetry properties passed in can override default properties

@cwhitten @vishwacsena Once these changes have been approved, I can raise an issue for composer to detail the changes needed in the runtime for telemetry.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/LUIS/LuisAdaptiveRecognizer.cs [hunk 1]
```diff
@@ -81,10 +81,13 @@ public LuisAdaptiveRecognizer()
...
-        /// Gets or sets a value indicating whether to log personal information that came from the user to telemetry.
+        /// Gets or sets the flag to determine if personal information should be logged in telemetry.
         /// </summary>
-        /// <value>If true, personal information is logged to Telemetry; otherwise the properties will be filtered.</value>
-        public bool LogPersonalInformation { get; set; } = false;
+        /// <value>
+        /// The flag to indicate in personal information should be logged in telemetry.
+        /// </value>
+        [JsonProperty("logPersonalInformation")]
+        public BoolExpression LogPersonalInformation { get; set; } = "=settings.telemetry.logPersonalInformation";
```
Comment (vishwacsena) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/LUIS/LuisAdaptiveRecognizer.cs:90]: For all these, we need an update to the LuisRecognizer component schema as well as re-gen the sdk.schema.

File: libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs [hunk 2]
```diff
@@ -70,22 +70,39 @@ public virtual async Task<T> RecognizeAsync<T>(DialogContext dialogContext, Acti
             return result;
         }
 
-        protected virtual Dictionary<string, string> FillRecognizerResultTelemetryProperties(RecognizerResult recognizerResult, Dictionary<string, string> telemetryProperties, ITurnContext turnContext = null)
+        protected virtual Dictionary<string, string> FillRecognizerResultTelemetryProperties(RecognizerResult recognizerResult, Dictionary<string, string> telemetryProperties, DialogContext dialogContext = null)
         {
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs:74]: Was this shipped in 4.8?  I would think this is a breaking change to change this signature.

File: libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs [hunk 3]
```diff
@@ -70,22 +70,39 @@ public virtual async Task<T> RecognizeAsync<T>(DialogContext dialogContext, Acti
...
-            telemetryProperties.Add("TopIntent", recognizerResult.Intents.Any() ? recognizerResult.Intents.First().Key : null);
-            telemetryProperties.Add("TopIntentScore", recognizerResult.Intents.Any() ? recognizerResult.Intents.First().Value?.ToString() : null);
-            telemetryProperties.Add("Intents", recognizerResult.Intents.Any() ? JsonConvert.SerializeObject(recognizerResult.Intents) : null);
-            telemetryProperties.Add("Entities", recognizerResult.Entities != null ? recognizerResult.Entities.ToString() : null);
-            telemetryProperties.Add("AdditionalProperties", recognizerResult.Properties.Any() ? JsonConvert.SerializeObject(recognizerResult.Properties) : null);
+            return properties;
+        }
+
+        protected void TrackRecognizerResult(DialogContext dialogContext, string eventName, Dictionary<string, string> telemetryProperties, Dictionary<string, double> telemetryMetrics)
+        {
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs:98]: protected methods are "public" for children, meaning there should be comment headers describing what this is.

File: libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs (hunk 2 above)
Comment (garypretty) [reply libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs:74]: It was. I had misunderstood the way the expression value worked, which means that the current implementation doesn't get the right application ID, so I think I do need the dialog context unless there is another way to get the value -> https://github.com/microsoft/botbuilder-dotnet/blob/0c8597686b81e035d0590687050c3ee8481ed2f3/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/LUIS/LuisAdaptiveRecognizer.cs#L140-L156  

If I do need DialogContext, given that adaptive isn't GA yet and we have not documented any of the telemetry bits yet, I think this is an acceptable change for us to make here. Thoughts?

File: libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs (hunk 3 above)
Comment (garypretty) [reply libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs:110]: Comments added. Thanks.

**Comment by garypretty** [comment]
Based on the review from the SDK team, we have decided that given the method with the altered signature is only currently used by the Adaptive recognizers, and is essentially only used internally, we will make the change.

**Pull request merged**